src/_cffi_src/openssl/callbacks.py

```diff
@@ -47,6 +47,7 @@
    using CPython APIs by Armin Rigo of the PyPy project.
 */
 
+#if Cryptography_HAS_LOCKING_CALLBACKS
```

Comment:
This value is not actually a preprocessor value so we can't use the preprocessor to check it. If you change this to `CRYPTOGRAPHY_OPENSSL_LESS_THAN_110` (which is what sets `Cryptography_HAS_LOCKING_CALLBACKS`) it will work properly.